pkg/util/collate/gb18030_chinese_ci.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collate
+
+import (
+	_ "embed"
+	"encoding/binary"
+
+	"github.com/pingcap/tidb/pkg/util/stringutil"
+)
+
+//go:embed gb18030_weight.data
+var gb18030WeightData []byte
```

Comment:
How did we get this `*.data` file? Is it generated by a program or downloaded from some places?